core-httpclient-impl/src/main/java/com/optimizely/ab/event/AsyncEventHandler.java

```diff
@@ -123,10 +136,17 @@ public AsyncEventHandler(int queueCapacity,
             .withEvictIdleConnections(1L, TimeUnit.MINUTES)
             .build();
 
+        NamedThreadFactory namedThreadFactory;
+        if (threadFactory == null) {
+            namedThreadFactory = new NamedThreadFactory("optimizely-event-dispatcher-thread-%s", true);
+        } else {
+            namedThreadFactory = new NamedThreadFactory("optimizely-event-dispatcher-thread-%s", true, threadFactory);
+        }
```

Comment:
```suggestion
        NamedThreadFactory namedThreadFactory = new NamedThreadFactory("optimizely-event-dispatcher-thread-%s", true, threadFactory);
```

Reply:
@wyhasany a question - replacing all "synchronized" with "lock" is required for Loom virtual threads or just for performance improvement?

Reply:
it is required for loom, as loom can't switch virtual thread when it is blocked by synchronized block.
> There are two scenarios in which a virtual thread cannot be unmounted during blocking operations because it is >pinned to its carrier:
>
>   1. When it executes code inside a synchronized block or method, or...
https://openjdk.org/jeps/444